context/context.go

```diff
@@ -181,15 +192,22 @@ type Response struct {
 // started means the response has sent out.
 func (w *Response) Write(p []byte) (int, error) {
 	w.Started = true
+	if !w.wroteHeader {
+		w.ResponseWriter.WriteHeader(w.Status)
+		//prevent multiple response.WriteHeader calls
+		w.wroteHeader = true
+	}
 	return w.ResponseWriter.Write(p)
 }
 
 // WriteHeader sends an HTTP response header with status code,
 // and sets `started` to true.
 func (w *Response) WriteHeader(code int) {
+	if w.Status > 0 {
+		return
+	}
 	w.Status = code
 	w.Started = true
-	w.ResponseWriter.WriteHeader(code)
```

Comment:
@astaxie I feel it's a little hard to fix this without a method like finishRequest (https://golang.org/src/net/http/server.go#L1243),but this method must be an expected method(FinishRequest).
So,what do you think about this?